content/insert.js

```diff
@@ -1,20 +1,54 @@
 function on_ok() {
   var latex = document.getElementById("tblatex-expr").value;
-  window.arguments[0](latex);
-}
-
-function on_cancel() {
-  window.close();
+  var autodpi = document.getElementById("autodpi-checkbox").checked;
+  var font_px = document.getElementById("fontpx").value;
+  window.arguments[0](latex, autodpi, font_px);
 }
 
 window.addEventListener("load", function (event) {
-  document.getElementById("tblatex-expr").value = window.arguments[1];
+  var template = window.arguments[1];
+  var selection = window.arguments[2];
+  populate(template, selection);
+  var autodpi = prefs.getBoolPref("autodpi");
+  document.getElementById("autodpi-checkbox").checked = autodpi;
+  update_ui(autodpi);
+  var font_px = prefs.getIntPref("font_px");
+  document.getElementById("fontpx").value = font_px;
 }, false);
 
+function on_reset() {
+  var template = prefs.getCharPref("template");
+  populate(template, null);
+}
+
+function on_autodpi() {
+  var autodpi = document.getElementById("autodpi-checkbox").checked;
+  update_ui(autodpi);
+}
+
 var prefs = Components.classes["@mozilla.org/preferences-service;1"]
   .getService(Components.interfaces.nsIPrefService)
   .getBranch("tblatex.");
 
-function on_reset() {
-  document.getElementById("tblatex-expr").value = prefs.getCharPref("template");
+function populate(template, selection) {
+  var marker = "__REPLACE_ME__";
+  var start = template.indexOf(marker);
+  if (start > -1 && selection) {
+    // Replace marker with selection
```

Comment:
in a future PR it'd be nice to provide an error message if the user accidentally deletes the __REPLACE_ME__ marker -- this has happened repeatedly in the past

Reply:
That's A Good Idea(TM)!